packages/app-cli/app/command-mkbook.ts

```diff
@@ -0,0 +1,68 @@
+const BaseCommand = require('./base-command').default;
+const { app } = require('./app.js');
+import { _ } from '@joplin/lib/locale';
+import BaseModel from '@joplin/lib/BaseModel';
+import Folder from '@joplin/lib/models/Folder';
+import { FolderEntity } from '@joplin/lib/services/database/types';
+
+class Command extends BaseCommand {
+	usage() {
+		return 'mkbook <new-notebook>';
+	}
+
+	description() {
+		return _('Creates a new notebook.');
+	}
+
+	options() {
+		return [
+			['-p, --parent <parent-notebook>', _('Create a new notebook under a parent notebook.')],
+		];
+	}
+
+	// validDestinationFolder check for presents and ambiguous folders
+	async validDestinationFolder(targetFolder: string) {
+
+		const destinationFolder = await app().loadItem(BaseModel.TYPE_FOLDER, targetFolder);
+		if (!destinationFolder) {
+			throw new Error(_('Cannot find: "%s"', targetFolder));
+		}
+
+		const destinationDups = await Folder.search({ titlePattern: targetFolder, limit: 2 });
+		if (destinationDups.length > 1) {
+			throw new Error(_('Ambiguous notebook "%s". Please use short notebook id instead - press "ti" to see the short notebook id', targetFolder));
+		}
+
+		return destinationFolder;
+	}
+
+	async saveAndSwitchFolder(newFolder: FolderEntity) {
+
+		const folder = await Folder.save(newFolder, { userSideValidation: true });
+		app().switchCurrentFolder(folder);
+
+	}
+
+	async action(args: any) {
+		const targetFolder = args.options.parent;
+
+		const newFolder: FolderEntity = {
+			title: args['new-notebook'],
+		};
+
+		if (targetFolder) {
+			if (typeof targetFolder === 'boolean') {
+				throw new Error(_('Missing required argument: parent-notebook'));
```

Comment:
Isn't it already enforced by the app? Argument in angled brackets are required, so it would already fail before you even reach this code

Reply:
Its true, it fail but with the output `Cannot find: "true" ` thats why i print a better error message

Reply:
In that case, whatever is generating this bad error message is what should be fixed, otherwise each command would need to have custom code to handle this

Reply:
@k33pn3xtlvl, I have now fixed the app so that it handles required flags correctly. Could you try to rebase, and see if it prints the message "Missing required flag value" when parent-notebook is not set?